### ISSUE TITLE
Thin out build matrix

### ISSUE THREAD

**Issue by rossabaker** [http4s/http4s#5475]
For JVM builds, we currently build the product of Scala {2.12,2.13,3} x Java {8,11,17}

It's all parallel, but we are rate limited in heavy PR waves.  If we cover the range of Javas and Scalas, the product doesn't seem so useful.  What if we restricted to:

* For Java 8, Scala {2.12, 2.13, 3}.  These are how we publish.
* For Scala 2.13, Java {8, 17}.  Oldest and newest.

9 jobs would go to 4.  Or 12 to 7, including the Scala.js jobs.

**Comment by armanbilge** [comment]
Makes sense to me. I would have included Java 11 for 2.13 as well, but happy to see it gone.

**Comment by rossabaker** [comment]
11 is fine for completeness, but only catches things that appeared and disappeared between 8 and 17.  Since it's still a major production version, and since it would no longer be multiplied by the number of Scalas, it's probably wise to keep.
